lib/config.js

```diff
@@ -229,13 +228,44 @@ function validateEnvConfig (prefix) {
   }
 }
 
+function deepFreeze (o) {
+  Object.freeze(o)
+
+  Object.getOwnPropertyNames(o).forEach(function (prop) {
+    if (o.hasOwnProperty(prop) &&
+        o[prop] !== null &&
+        (o[prop].constructor === Object || typeof o[prop] === 'function') &&
```

Comment:
You're right. `o[prop] !== null` Should be `o[prop] != null` _or_ `o[prop] !== null && o[prop] !== undefined`